pages/api/login.ts

```diff
@@ -0,0 +1,26 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { withIronSessionApiRoute } from 'iron-session/next'
+import { sessionOptions } from '../../libs/session'
+import type { IUser } from '../../models/user.model'
+import { userRepository } from './../../libs/user-repository'
+
+async function loginRoute(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const user = userRepository.checkAuth(req.body) as IUser
```

Comment:
```suggestion
    const user = userRepository.checkAuth(req.body)
```
should defined return type in `repository`